zio-http/shared/src/main/scala/zio/http/Route.scala

```diff
@@ -51,13 +51,13 @@ sealed trait Route[-Env, +Err] { self =>
    * into one that does handle its errors.
    */
   final def handleError(f: Err => Response)(implicit trace: Trace): Route[Env, Nothing] =
-    self.handleErrorCause(Response.fromCauseWith(_)(f))
+    self.handleErrorCauseZIO(c => ErrorResponseConfig.configRef.get.map(Response.fromCauseWith(c, _)(f)))
 
   final def handleErrorZIO(f: Err => ZIO[Any, Nothing, Response])(implicit trace: Trace): Route[Env, Nothing] =
     self.handleErrorCauseZIO { cause =>
       cause.failureOrCause match {
         case Left(err)    => f(err)
-        case Right(cause) => ZIO.succeed(Response.fromCause(cause))
+        case Right(cause) => ErrorResponseConfig.configRef.get.map(Response.fromCause(cause, _))
```

Comment:
Not as nice, but it requires 1 FlatMap less:
```suggestion
        case Right(cause) => ErrorResponseConfig.configRef.getWith(c => ZIO.succeed(Response.fromCause(cause, c)))
```

There are a few other places this can be applied to

Reply:
Still no e enough